test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs

```diff
@@ -3071,6 +3071,45 @@ public Task TestPacManGetUnrestoredPackagesByDependencyOrder()
                 Assert.Equal(NuGetProjectActionType.Install, packageActions[5].NuGetProjectActionType);
                 Assert.Equal(singlePackageSource, packageActions[5].SourceRepository.PackageSource.Source);
 
+                string projectName;
+                if (!msBuildNuGetProject.TryGetMetadata(NuGetProjectMetadataKeys.Name, out projectName))
+                {
+                    projectName = string.Empty;
```

Comment:
there shouldn't be a fallback here or down in the event handlers, for a test this property should always exist